### DISCUSSION TITLE
Visualizing Otel data

### DISCUSSION THREAD

**Discussion by rogeralsing** [open-telemetry/opentelemetry-demo#1453] [Show and tell]
I'm working on a visualization tool for Otel data.
(https://github.com/asynkron/TraceLens for anyone interested)

Yesterday I tried it on the OpenTelemetry Demo application.

I would love to hear some feedback on how you think this matches the actual architecture of the various services.

<img width="1384" alt="Skärmavbild 2024-03-13 kl  12 19 03" src="https://github.com/open-telemetry/opentelemetry-demo/assets/647031/26b182d2-8339-41f6-b686-f3f4a08f97f1">

Operation level details:
<img width="1378" alt="Skärmavbild 2024-03-13 kl  12 23 16" src="https://github.com/open-telemetry/opentelemetry-demo/assets/647031/26709848-f13b-46d5-b9eb-3b47c7d91c07">

I also have a question regarding the Kafka part of the demo app.
Does it lack instrumentation when publishing _to_ Kafka?
I see the Publish action as a span. but there are no parent spans leading to it. I would have expected one of the services to have generated a span that leads to the Publish action.

**Comment by puckpuck** [comment]
I'm both happy you did this and sad you did as well since you exposed an ugly truth about the Otel demo and the `instrumentation-fs` library with NodeJS that we should address.

This entire section in the frontend box should not be there. I'm certain there are options we can do to make that happen.

![Screenshot 2024-03-18 at 10 42 33 PM](https://github.com/open-telemetry/opentelemetry-demo/assets/1296118/60a7d555-6aa3-4631-88f7-4c90028a4d0b)
